doc/design/state-machine.md

```diff
@@ -0,0 +1,185 @@
+# HiveD State Machines
+
+This document presents the state machines of HiveD, and explains the life cycles of jobs and resources in HiveD.
+We will describe the state machines of our scheduling unit, Affinity Group, and that of our resource unit, Cell, respectively.
+
+## Affinity Group (AG) State Machine
+
+An affinity group (AG) is a set of gang-scheduled pods. It is the basic scheduling unit in HiveD. The figure below shows the state machine of an affinity group.
+
+Note that the AG state machine has interactions with the cell state machine (elaborated later). In our design, AGs influence each other only via cell: an event of an AG may trigger an event for one of its cells, and the influenced cell may trigger another event for another AG related to the same cell.
```

Comment:
In our design, AGs influence each other only via their shared cells, for example, an event sent to an AG may trigger another event sent to a cell, and the cell may further trigger another event sent to another AG which is current assosiated with the cell.